option.go

```diff
@@ -78,7 +93,10 @@ type tracerProviderOption struct {
 
 func (o *tracerProviderOption) apply(c *config) {
 	if o.provider != nil {
-		c.tracerProvider = o.provider
+		c.tracer = o.provider.Tracer(
+			instrumentationName,
+			trace.WithInstrumentationVersion(semanticVersion),
+		)
```

Comment:
traceprovider was only used with the same instrument naming so we can simplify by just initialising a tracer here